### DISCUSSION TITLE
Feature: dev-only warning for resources

### DISCUSSION THREAD

**Discussion by lxsmnsyc** [solidjs/solid#1034] [Ideas]
Currently, you can safely read resources or mount lazy components without even using `<Suspense>` however I believe this pattern should be enforced, regardless of whether or not `<Suspense>` is optional. 

The idea is to introduce a utility function much like `enableScheduling`, let's name it `enableSuspenseWarnings`, that the user can use to raise warnings for whether a resource has been accessed outside a Suspense boundary.

There are other benefits like when using `renderToStringAsync` or `pipeToNodeWriteable`, which may force users to use `<Suspense>` for the SSR methods to properly render.

**Comment by ryansolid** [comment]
I didn't like this idea at first but I'm coming around. There might be other warnings like this to consider.

**Comment by ryansolid** [comment]
This requires more thought and alignment with resource strategy in general. Moving to discussions.

**Reply by lxsmnsyc** [reply]
I've just read this. Any reason why it should consider the requirements? is it because resources do not have any idea about Suspense boundaries?

**Reply by ryansolid** [reply]
I was just commenting on the focus resources are getting right now and that we probably need to make some decisions here. There was a proposal in another discussion around separate resource from Suspense mechanism. I just saw the issue a bit out of place and should be part of any plan we make here.

**Reply by lxsmnsyc** [reply]
Right. I think there should be a thread that shows the collective plan for the development of resource.
